pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -214,23 +233,82 @@ def default_qnode_transform(self, qnode, targs, tkwargs):
         if self.expand_transform:
             qnode.add_transform(
                 TransformContainer(
-                    self._expand_transform, targs, tkwargs, use_argnum=self._use_argnum_in_expand
+                    self._expand_transform,
+                    args=targs,
+                    kwargs=tkwargs,
+                    use_argnum=self._use_argnum_in_expand,
                 )
             )
         qnode.add_transform(
             TransformContainer(
                 self._transform,
-                targs,
-                tkwargs,
-                self._classical_cotransform,
-                self._is_informative,
-                self._final_transform,
+                args=targs,
+                kwargs=tkwargs,
+                classical_cotransform=self._classical_cotransform,
+                plxpr_transform=self._plxpr_transform,
+                is_informative=self._is_informative,
+                final_transform=self._final_transform,
             )
         )
         return qnode
 
+    def plxpr_transform(self, primitive, tracers, params, targs, tkwargs, state):
+        """Function for processing primitives to transform PLxPR.
+        Args:
+            primitive (jax.core.Primitive): Primitive to transform
+            tracers (Sequence[jax.core.Tracer]): Input tracers to the primitive
+            params (dict): Dictionary containing keyword arguments/metadata for the primitive
+            targs (Sequence[Any]): Arguments for the transform
+            tkwargs (dict): Keyword arguments for the transform
+            state (dict): Dictionary containing auxiliary information about the environment/state
+                needed to apply the transform
+        Returns:
+            Any: The results of the transformed primitive
+        """
+        # Implemented this way rather than using a property so that the correct docstring is used
+        return self._plxpr_transform(primitive, tracers, params, targs, tkwargs, state)
+
+    def _capture_callable_transform(self, qfunc, targs, tkwargs):
+        """Apply the transform on a quantum function when program capture is enabled"""
+
+        @functools.wraps(qfunc)
+        def qfunc_transformed(*args, **kwargs):
+            import jax  # pylint: disable=import-outside-toplevel
+
+            nonlocal qfunc
```

Comment:
?

Reply:
I was getting errors about missing variables without it.

Reply:
Maybe try `if isinstance(qfunc, qml.QNode) qfunc else qfunc = `

Reply:
I removed it altogether. The intention for qnode-related validation was to support the existing execution pipeline, but it is already a weird hybrid of old and new capture, so let's keep it restricted for now.